main.go

```diff
@@ -51,57 +49,46 @@ func main() {
 	)
 	defer cancel()
 
-	logrus.SetFormatter(&nested.Formatter{})
-	ctx = log.WithFields(ctx, map[string]interface{}{"cmd": os.Args[:1]})
-	ctx = log.WithLog(ctx, logruslogger.New(ctx))
+	_, sLogger, span, sFinish := spanlogger.FromContext(ctx, "cmd-nsmgr")
+	defer sFinish()
+	_, lLogger, lFinish := logruslogger.FromSpan(ctx, span, "cmd-nsmgr")
+	defer lFinish()
+	logger := log.Combine(sLogger, lLogger)
 
 	// ********************************************************************************
 	// Debug self if necessary
 	// ********************************************************************************
 	if err := debug.Self(); err != nil {
-		log.FromContext(ctx).Infof("%s", err)
+		logger.Infof("%s", err)
 	}
 
 	// ********************************************************************************
 	// Configure open tracing
 	// ********************************************************************************
 	// Enable Jaeger
 	log.EnableTracing(true)
-	jaegerCloser := jaeger.InitJaeger(ctx, "nsmgr")
+	jaegerCloser := jaeger.InitJaeger(log.WithLog(ctx, logger), "nsmgr")
 	defer func() { _ = jaegerCloser.Close() }()
-	traceCtx, finish := withTraceLogger(ctx, "nsmgr")
 
 	// Get cfg from environment
 	cfg := &config.Config{}
 	if err := envconfig.Usage("nsm", cfg); err != nil {
-		logrus.Fatal(err)
+		logger.Fatal(err)
 	}
 	if err := envconfig.Process("nsm", cfg); err != nil {
-		logrus.Fatalf("error processing cfg from env: %+v", err)
+		logger.Fatalf("error processing cfg from env: %+v", err)
 	}
 
-	logrus.Infof("Using configuration: %v", cfg)
+	logger.Infof("Using configuration: %v", cfg)
 
 	level, err := logrus.ParseLevel(cfg.LogLevel)
 	if err != nil {
-		logrus.Fatalf("invalid log level %s", cfg.LogLevel)
+		logger.Fatalf("invalid log level %s", cfg.LogLevel)
 	}
 	logrus.SetLevel(level)
 
-	// Startup is finished
-	finish()
```

Comment:
Hmm, this can be a major. Can we run here `sFinish()`?

Reply:
Yes, we can re-recreate the span logger for the next operation.

Reply:
ok, done